package data;

message PbRouteId {
 required uint64 value = 1;
}

message PbOneLineText {
 required string text = 1;
}

message PbDate {
 required uint32 year = 1;
 required uint32 month = 2;
 required uint32 day = 3;
}

message PbTime {
 required uint32 hour = 1;
 required uint32 minute = 2;
 required uint32 seconds = 3;
 optional uint32 millis = 4 [default = 0];
}

message PbSystemDateTime {
 required PbDate date = 1;
 required PbTime time = 2;
 required bool trusted = 3;
}

message PbLocation {

 enum Fix {
  FIX_NONE = 0;
  FIX_2D = 1;
  FIX_3D = 2;
 }
 required double latitude = 1;
 required double longitude = 2;
 optional PbSystemDateTime timestamp = 3;
 optional PbLocation.Fix fix = 4;
 optional uint32 satellites = 5 [default = 0];
}

message PbRoutePoint {
 required sint32 x_offset = 1;
 required sint32 y_offset = 2;
 optional uint32 time_offset = 3;
 optional sint32 z_offset = 4;
}

message PbPlannedRoute {
 required PbRouteId route_id = 1;
 required PbOneLineText name = 2;
 optional float length = 3;
 optional PbLocation start_location = 4;
 optional float start_altitude = 5;
 repeated PbRoutePoint point = 6;
}

